bridge-token-factory/tests/token_transfer.rs

```diff
@@ -63,161 +84,218 @@ impl From<Proof> for AugmentedProof {
     .to_log_entry_data();
 
-    call_borsh!(user, factory_id.deposit(proof)).assert_success();
-
-    let alice_balance: String =
-        call_json!(user, token_account_id.ft_balance_of({"account_id": ALICE.to_string()}))
-            .unwrap_json();
-    assert_eq!(alice_balance, "1000");
-
-    let total_supply: String = call_json!(user, token_account_id.ft_total_supply({})).unwrap_json();
-    assert_eq!(total_supply, "1000");
-
-    call_json!(alice, token_account_id.withdraw({
-                "amount" : "100",
-                "recipient" : SENDER_ADDRESS.to_string()}), deposit=1)
-    .assert_success();
-
-    let alice_balance: String =
-        call_json!(user, token_account_id.ft_balance_of({"account_id": ALICE.to_string()}))
-            .unwrap_json();
-    assert_eq!(alice_balance, "900");
-
-    let total_supply: String = call_json!(user, token_account_id.ft_total_supply({})).unwrap_json();
-    assert_eq!(total_supply, "900");
+    assert!(&rt
+        .block_on(
+            user.call(&worker, factory.id(), "deposit")
+                .deposit(parse_near!("50 N"))
+                .max_gas()
+                .args(proof.try_to_vec().unwrap())
+                .transact()
+        )
+        .unwrap()
+        .is_success());
+
+    let alice_balance: String = std::str::from_utf8(
+        &rt.block_on(
+            worker.view(
+                &token_account_id.parse().unwrap(),
+                "ft_balance_of",
+                json!({"account_id": user.id().to_string()})
+                    .to_string()
+                    .into_bytes(),
+            ),
+        )
+            .unwrap()
+            .result,
+    ).unwrap().to_string();
+    assert_eq!(alice_balance, "\"1000\"");
+
+    let total_supply: String = std::str::from_utf8(
+        &rt.block_on(worker.view(
+            &token_account_id.parse().unwrap(),
+            "ft_total_supply",
+            json!({}).to_string().into_bytes(),
+        )).unwrap().result,
+    ).unwrap().to_string();
+    assert_eq!(total_supply, "\"1000\"");
+
+    assert!(&rt.block_on(
+        user.call(&worker, &token_account_id.parse().unwrap(), "withdraw")
+            .max_gas()
+            .deposit(1)
+            .args(
+                json!({
+                        "amount" : "100",
+                        "recipient" : SENDER_ADDRESS.to_string()
+                    })
+                    .to_string()
+                    .into_bytes(),
+            )
+            .transact(),
+    ).unwrap().is_success());
+
+    let alice_balance: String = std::str::from_utf8(
+        &rt.block_on(
+            worker.view(
+                &token_account_id.parse().unwrap(),
+                "ft_balance_of",
+                json!({"account_id": user.id().to_string()})
+                    .to_string()
+                    .into_bytes(),
+            ),
+        )
+            .unwrap()
+            .result,
+    ).unwrap().to_string();
+    assert_eq!(alice_balance, "\"900\"");
```

Comment:
Let's calculate this value instead of using a hardcoded one.

Reply:
fixed: https://github.com/aurora-is-near/rainbow-token-connector/pull/159/commits/3090dbc3011165b6f895bbe177cf3d44d5549d99